canvas.h

```diff
@@ -125,12 +125,9 @@ class Canvas : public ICanvas
 inline void to_json(nlohmann::json& j, const ICanvas& canvas) 
 {
     // Serialize the features
-    vector<nlohmann::json> serializedFeatures;
-    for (const auto& feature : canvas.Features()) {
-        if (feature) {
-            serializedFeatures.push_back(*feature); // Dereference the shared pointer
-        }
-    }
+    vector<nlohmann::json> jsonFeatures;
+    for (const auto& feature : canvas.Features())
+        jsonFeatures.push_back(*feature); // Dereference the shared pointer
```

Comment:
The "nullptr" check is not necessary because nullptr features cannot be added to a canvas in the first place.